src/quark-core/src/QuarkWallet.sol

```diff
@@ -455,6 +456,15 @@ contract QuarkWallet is IERC1271 {
             oldActiveSubmissionToken := tload(activeSubmissionTokenSlot)
             oldCallback := tload(callbackSlot)
 
+            // Prevent nested operations coming from an outside caller (i.e. not the Quark wallet itself)
+            if and(iszero(eq(oldActiveScript, 0)), iszero(eq(caller(), address()))) {
```

Comment:
We check against the `activeScript` here, but we really could also check against `submissionToken` since that is guaranteed to be non-zero during a nested operation. `activeScript` should also be non-zero during a nested operation since the zero address has empty code and will revert from the check above (L437).

Reply:
Embrace the YUL.

```
and(iszero(eq(oldActiveScript, 0)), iszero(eq(caller(), address())))

if ( ( oldActiveScript == 0 ) == FALSE ) AND ( ( msg.sender == address[this] ) == FALSE )
if oldActiveScript != 0 and msg.sender != address[this]

DeMorgan's Law:

unless (oldActiveScript == 0 or msg.sender == address[this]) {
  revert()
}
```

My work through of the logic there.

Reply:
Correct, that logic checks out